pkg/storage/db.go

```diff
@@ -293,23 +268,66 @@ func (db *DB) runQuery(ctx context.Context, query string, args ...interface{}) (
 
 	defer rows.Close()
 
-	networkChangeEvents := make([]domain.NetworkChangeEvent, 0)
+	cloudAssetDetails := make([]domain.CloudAssetDetails, 0)
+
+	tempMap := make(map[string]*domain.CloudAssetDetails)
+
 	for rows.Next() {
-		var row domain.NetworkChangeEvent
-		var bytes []byte
+		var row domain.CloudAssetDetails
+
+		var metaBytes []byte
 		var hostname sql.NullString
-		err = rows.Scan(&row.ResourceID, &row.IPAddress, &hostname, &row.IsPublic, &row.IsJoin, &row.Timestamp, &row.AccountID, &row.Region, &row.Type, &bytes)
+		var ipAddress sql.NullString
+		var isPublic sql.NullBool
+		var isJoin sql.NullBool
+		var timestamp time.Time
+
+		err = rows.Scan(&row.ResourceID, &ipAddress, &hostname, &isPublic, &isJoin, &timestamp, &row.AccountID, &row.Region, &row.ResourceType, &metaBytes)
+
 		if err == nil {
-			if bytes != nil {
+			if metaBytes != nil {
 				var i map[string]string
-				_ = json.Unmarshal(bytes, &i) // we already checked for nil, and the DB column is JSONB; no need for err check here
+				_ = json.Unmarshal(metaBytes, &i) // we already checked for nil, and the DB column is JSONB; no need for err check here
 				row.Tags = i
 			}
+			if tempMap[row.ResourceID] == nil {
+				tempMap[row.ResourceID] = &row
+			}
+			found := false
 			if hostname.Valid {
-				row.Hostname = hostname.String
-
+				for _, val := range tempMap[row.ResourceID].Hostnames {
+					if strings.EqualFold(val, hostname.String) {
+						found = true
+						break
+					}
+				}
+				if !found {
+					tempMap[row.ResourceID].Hostnames = append(tempMap[row.ResourceID].Hostnames, hostname.String)
+				}
+			}
+			found = false
+			if ipAddress.Valid {
+				var ipAddresses *[]string
+				if isPublic.Bool { // no need for .Valid check; DB table column guarantees non-null
```

Comment:
If I understand the comment, is it necessary to use `sql.NullBool` for this?

Reply:
You're right, I didn't need that extra bit.  Good catch.